Providers/RustSyncManager.swift

```diff
@@ -0,0 +1,553 @@
+            }
+
+            accountManager.getAccessToken(scope: OAuthScope.oldSync) { result in
+                guard let accessTokenInfo = try? result.get(),
+                      let key = accessTokenInfo.key else {
+                    deferred.fill(Maybe(failure: ScopedKeyError()))
+                    return
+                }
+
+                accountManager.getTokenServerEndpointURL { result in
+                    guard case .success(let tokenServerEndpointURL) = result else {
+                        deferred.fill(Maybe(failure: NoTokenServerURLError()))
+                        return
+                    }
+
+                    self.getEnginesAndKeys(engines: engines).upon { result in
+                        guard let (rustEngines, localEncryptionKeys) = result
+                            .successValue else {
+                            deferred.fill(Maybe(failure: EngineAndKeyRetrievalError()))
+                            return
+                        }
+                        let params = SyncParams(
+                            reason: why,
+                            engines: SyncEngineSelection.some(engines: rustEngines),
+                            enabledChanges: self.getEngineEnablementChangesForAccount(),
+                            localEncryptionKeys: localEncryptionKeys,
+                            authInfo: SyncAuthInfo(
+                                kid: key.kid,
+                                fxaAccessToken: accessTokenInfo.token,
+                                syncKey: key.k,
+                                tokenserverUrl: tokenServerEndpointURL.absoluteString),
+                            persistedState:
+                                self.prefs
+                                    .stringForKey(PrefsKeys.RustSyncManagerPersistedState),
+                            deviceSettings: DeviceSettings(
+                                fxaDeviceId: device.id,
+                                name: device.displayName,
+                                kind: self.toSyncManagerDeviceType(
+                                    deviceType: device.deviceType)))
+
+                        self.beginSyncing()
+                        self.syncManagerAPI.sync(params: params) { syncResult in
+                            guard syncResult.status == ServiceStatus.ok else {
+                                deferred.fill(Maybe(success: syncResult))
+                                self.endSyncing(syncResult)
+                                return
+                            }
+
+                            // Save the persisted state
+                            self.prefs
+                                .setString(syncResult.persistedState,
+                                           forKey: PrefsKeys.RustSyncManagerPersistedState)
+
+                            let declinedEngines = String(describing: syncResult.declined ?? [])
+                            let telemetryData = syncResult.telemetryJson ??
```

Comment:
ditto here - we want to report failures too.

Reply:
That `telemetryData` variable is for the logging. The telemetry is reported in the `endSyncing` function. Worth noting that the temp telemetry doesn't report status, which is just another reason for it to go.